### PR TITLE
PairCorrEstimator fix

### LINKED ISSUE #396: Tests of gofr estimator needed
SoA implemenation is added in #395 but the correctness is not tested due to the lack of tests.

### PR DESCRIPTION
**Pull request by jptowns**
Should eliminate merge conflict from previous (now closed) PR.
Includes changes to PairCorrEstimator to exactly compute norm_factor, as well as the unit test and CMake update.
replaces #2066 and #2067
Closes #2035 
Closes #396

### CODE REVIEW

File: src/QMCHamiltonians/PairCorrEstimator.cpp [hunk 1]
```diff
@@ -28,17 +28,22 @@ PairCorrEstimator::PairCorrEstimator(ParticleSet& elns, std::string& sources)
   UpdateMode.set(COLLECTABLE, 1);
   num_species = elns.groups();
   n_vec.resize(num_species, 0);
-  for (int i(0); i < num_species; i++)
-    n_vec[i] = 1.0 / (elns.last(i) - elns.first(i));
+  for ( int i=0; i<num_species; i++ )
+    {
```
Comment (ye-luo) [inline src/QMCHamiltonians/PairCorrEstimator.cpp]: Could you remove the curly brackets and save two lines?

File: src/QMCHamiltonians/PairCorrEstimator.cpp [hunk 2]
```diff
@@ -28,17 +28,22 @@ PairCorrEstimator::PairCorrEstimator(ParticleSet& elns, std::string& sources)
...
+  // use the simulation cell radius if any direction is periodic
+  if ( elns.Lattice.SuperCellEnum )
   {
     Dmax   = elns.Lattice.WignerSeitzRadius;
     Volume = elns.Lattice.Volume;
   }
-  else
-    Volume = 1.0; //open bcs
+  else  // Open BC's
+    {
```
Comment (ye-luo) [inline src/QMCHamiltonians/PairCorrEstimator.cpp]: Could you remove the curly brackets and save two lines?

File: src/QMCHamiltonians/PairCorrEstimator.cpp [hunk 3]
```diff
@@ -47,67 +52,77 @@ PairCorrEstimator::PairCorrEstimator(ParticleSet& elns, std::string& sources)
   //gof_r_prefix.push_back(h.str());
   std::map<int, int> pair_map;
   int npairs = 0;
-  for (int i = 0; i < num_species; ++i)
-    for (int j = i; j < num_species; ++j, ++npairs)
+  for ( int i=0; i<num_species; ++i )
     {
```
Comment (ye-luo) [inline src/QMCHamiltonians/PairCorrEstimator.cpp]: Could you remove the curly brackets and save two lines?

File: src/QMCHamiltonians/PairCorrEstimator.cpp (hunk 2 above)
Comment (jptowns) [reply src/QMCHamiltonians/PairCorrEstimator.cpp]: I initially submitted two PR's. One included only the unit test #2066 , while the other provided a fix to PairCorrEstimator to make the unit test pass #2067 . 

However, it seems my fork was behind mainline develop, so there were some issues with both. I updated my fork and re-implemented my changes in this new PR, which includes both the unit test and the fix. 

I've just pushed changes that address your comments above about brackets, as well as your additions in #2066 about the instantiation of PairCorrEstimator.

Apologies if I've bungled the process!

**Comment by ye-luo** [comment]
Please run the clang-format. You new code includes tabs which is problematic.

**Comment by ye-luo** [comment]
Okay to test

**Pull request merged**